### PR TITLE
[C][Client] Treat "null" as a valid value for a field of a JSON map

### PR DESCRIPTION
**Pull request by ityuhui**
Hi @wing328 @zhemant @michelealbano

When a field of a JSON map is null, e.g. in below example, `annotations` of  `metadata` is null

```json
{
    "apiVersion": "v1",
    "kind": "ServiceAccount",
    "metadata": {
        "name": "test-sa",
        "namespace": "smoking-test",
        "annotations": null
    }
}
```

`*_parseFromJSON` will treat `annotations` as invalid/error, and free the total memory of `metadata` 

The detail is at https://github.com/kubernetes-client/c/issues/138

This PR treats "null" as a valid value for a field of a JSON map.

### PR checklist
 
- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [X] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [X] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.1.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [X] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by ityuhui** [comment]
The sample build succeeded.

```bash
make
Scanning dependencies of target openapi_petstore
[  6%] Building C object CMakeFiles/openapi_petstore.dir/src/list.c.o
[ 13%] Building C object CMakeFiles/openapi_petstore.dir/src/apiKey.c.o
[ 20%] Building C object CMakeFiles/openapi_petstore.dir/src/apiClient.c.o
[ 26%] Building C object CMakeFiles/openapi_petstore.dir/external/cJSON.c.o
[ 33%] Building C object CMakeFiles/openapi_petstore.dir/model/object.c.o
[ 40%] Building C object CMakeFiles/openapi_petstore.dir/model/api_response.c.o
[ 46%] Building C object CMakeFiles/openapi_petstore.dir/model/category.c.o
[ 53%] Building C object CMakeFiles/openapi_petstore.dir/model/order.c.o
[ 60%] Building C object CMakeFiles/openapi_petstore.dir/model/pet.c.o
[ 66%] Building C object CMakeFiles/openapi_petstore.dir/model/tag.c.o
[ 73%] Building C object CMakeFiles/openapi_petstore.dir/model/user.c.o
[ 80%] Building C object CMakeFiles/openapi_petstore.dir/api/PetAPI.c.o
[ 86%] Building C object CMakeFiles/openapi_petstore.dir/api/StoreAPI.c.o
[ 93%] Building C object CMakeFiles/openapi_petstore.dir/api/UserAPI.c.o
[100%] Linking C shared library libopenapi_petstore.so
[100%] Built target openapi_petstore
```

**Comment by wing328** [comment]
Thanks for the fix and the test. Will merge tomorrow if no feedback from anyone.

(I'll try to add some tests in another PR after merging this one)

**Review by zhemant — APPROVED** [review]

**Comment by ityuhui** [comment]
Thanks @zhemant 

I'll try to add the condition `{{^isNullable}}` or `{{#isNullable}}` to the code change.

**Comment by ityuhui** [comment]
I looked at the log to generate clients and found all `isNullable` are`false` by default.
That is, Openapi-Generator Core treats all variables as `"isNullable" : false` if they don't explictly specify `nullable: true`

But as described at https://github.com/kubernetes-client/c/issues/138#issue-1350294784:

_Original yaml which could be well processed by kubectl:_

Kubectl supports this behavior (`annotations: null`) 
Actually the the version of OpenAPI spec for Kubernetes is OpenAPI V2.  But "nullable" is defined only by OpenAPI 3.0 https://swagger.io/docs/specification/data-models/data-types/

So now we cannot add `{{^isNullable}}` or we will break the behavior of Open API V2 

I recommend merge this PR now and fix the issue when we meet it in future. @zhemant

**Comment by wing328** [comment]
In openapi 2.0 spec, one can use `x-nullabe`

> I'll try to add the condition {{^isNullable}} or {{#isNullable}} to the code change.

Good idea.

**Comment by ityuhui** [comment]
Thank you @wing328 

Do you mean I need submit an issue or PR to add `x-nullable` to kubernetes spec https://github.com/kubernetes/kubernetes/blob/master/api/openapi-spec/swagger.json ?  I wonder if the kubernetes community accepts this change.

**Comment by wing328** [comment]
No, I'm not sure about the current status of https://github.com/kubernetes/kubernetes/blob/master/api/openapi-spec/swagger.json. I though there were plans to migrate it to OpenAPI 3.x. I can find some v3 spec under https://github.com/kubernetes/kubernetes/tree/master/api/openapi-spec/v3 but not sure if these are ready for production usage.

I just want to share there's an unofficial way to describe nullable properties/parameters in openpai 2.0 spec.

**Pull request merged**
